### PR TITLE
Side Nav Styles

### PR DESCRIPTION
**Pull request by DaveVanVliet**
[ADO 13009](https://dev.azure.com/AMA-Ent/AMA-Ent/_workitems/edit/13009) 🎨  

This PR has a corresponding set of API changes which contain some markup changes. (https://github.com/amaabca/api/pull/630)

I left the component block name as `side-nav` for now in an effort to try and preserve class names for testing purposes. If as a design team we don't think this is worthwhile, I can go back in and revise the name to be `top-nav`.

Some suggestions on what to review (Changes are on staging, but will **only have the corresponding API changes in the registries application**):
- A logged in member
- A logged in non-member
- Someone registering for a membership - (This _will_ appear wrong, and will need to be fixed within the membership application.)

To be completed by the person who opens the PR. Use strikethroughs for items that are not applicable in list.
- [x] I have reviewed my own PR to check syntax & logic, removed unnecessary/old code, made sure everything aligns with our style guide and BEM.
- [ ] I've added new components to the style guide (or have a PR in to add them).
- [ ] Any applicable version numbers have been updated.
- [x] I've tested on mobile, tablet, and desktop, as well as across all of the browsers we support.
- [x] I've proof-read all text for legibility, proper grammar, punctuation, and capitalization (titlecase).
- [x] I have written a detailed PR message explaining what is being changed and why
- [x] I’ve linked to any relevant VSTS tickets
- [x] I’ve added the appropriate review symbols to my PR - (elephant) for dev review, (art) for design

### CODE REVIEW

File: app/assets/stylesheets/v3/layout_components/side_nav.scss [hunk 1]
```diff
@@ -1,11 +1,29 @@
+/* Desktop Header/Side-Nav */
+
 .side-nav {
+  @include grid-column(12,0);
```
Comment (sineade) [inline app/assets/stylesheets/v3/layout_components/side_nav.scss:4]: same question as above about the marketing site width

File: app/assets/stylesheets/v3/layout_components/siteheader.scss [hunk 2]
```diff
@@ -81,7 +85,7 @@
 }
 
 .top-nav-offset {
-  margin-top: $top-bar-height;
+  margin-top: $top-bar-height + $side-nav-height;
```
Comment (sineade) [inline app/assets/stylesheets/v3/layout_components/siteheader.scss:88]: on mobile, this seems a bit tall.

File: app/assets/stylesheets/v3/layout_components/sections.scss [hunk 3]
```diff
@@ -291,7 +291,7 @@
 
 .main-container {
   // using grid-column in place of large-10 so that we can set the gutters to 0
-  @include grid-column(10,0);
+  @include grid-column(12,0);
```
Comment (sineade) [inline app/assets/stylesheets/v3/layout_components/sections.scss:294]: should we try to imitate the width of the marketing site more?

File: app/assets/stylesheets/v3/layout_components/side_nav.scss [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
   background: $white;
-  @include grid-column(2,0);
+  box-shadow: $base-box-shadow;
+  height: $side-nav-height;
+  position: fixed;
```
Comment (sineade) [inline app/assets/stylesheets/v3/layout_components/side_nav.scss:8]: I noticed the marketing site's navigation isn't fixed. How did you decide to make ours that way?

File: app/assets/stylesheets/v3/layout_components/sections.scss (hunk 3 above)
Comment (DaveVanVliet) [reply app/assets/stylesheets/v3/layout_components/sections.scss:294]: I think we had a quick call about this last week and decided to approach this once v2 comes around. The biggest challenge was that we need a full-width page header, but the rest of the page content should be inset with their custom page widths, and that would require custom code in each application.

File: app/assets/stylesheets/v3/layout_components/side_nav.scss (hunk 4 above)
Comment (DaveVanVliet) [reply app/assets/stylesheets/v3/layout_components/side_nav.scss:8]: Their main navigation isn't fixed, but their sub-navigation is (The blue one). I guess I figured that we rely pretty heavily on a member having quick access to our side nav (now top nav) and it seemed reasonable to replace one sticky navigation with another?

File: app/assets/stylesheets/v3/layout_components/siteheader.scss (hunk 2 above)
Comment (DaveVanVliet) [reply app/assets/stylesheets/v3/layout_components/siteheader.scss:88]: I'll take a look!

File: app/assets/stylesheets/v3/layout_components/side_nav.scss (hunk 4 above)
Comment (sineade) [reply app/assets/stylesheets/v3/layout_components/side_nav.scss:8]: sounds reasonable to me!

File: app/assets/stylesheets/v3/layout_components/sections.scss (hunk 3 above)
Comment (sineade) [reply app/assets/stylesheets/v3/layout_components/sections.scss:294]: ah okay, sorry I either forgot or wasn't there for that. Sounds good.

File: app/assets/stylesheets/v3/layout_components/side_nav.scss [hunk 5]
```diff
@@ -31,61 +49,125 @@
...
   &__content {
-    padding: $base-padding;
+    display: flex;
   }
 
-  &__list {
+  &__content-container {
+    @extend .side-nav__content;
+    flex-grow: 1;
+    justify-content: space-evenly;
```
Comment (kaytcampbell) [inline app/assets/stylesheets/v3/layout_components/side_nav.scss:68]: https://caniuse.com/#search=space-evenly This doesn't work in IE. I wonder if there is a fallback/other solution we can use for IE?
image

File: app/assets/stylesheets/v3/layout_components/siteheader.scss (hunk 2 above)
Comment (DaveVanVliet) [reply app/assets/stylesheets/v3/layout_components/siteheader.scss:88]: You're right, I've updated this for mobile.

**Comment by DaveVanVliet** [comment]
@kaytcampbell 
>can we make the font sizes smaller for the links in the very top blue bar? like 16px? i think that's what they were in the mocks.
can we make the arrow beside the logged in person's name white instead of blue? it's hard to see.

For these two points, since those would be changes to the top nav, would you be ok with them occurring on a separate PR? (I haven't taken a crack at styling the top-top-nav yet and I think those styles could be done completely independent from these sidebar changes.

**Comment by DaveVanVliet** [comment]
@kaytcampbell @sineade 👋 

I've:
- Reduced the size of the mobile header.
- Updated the flexbox attributes to more ie friendly values (And tested them on ie)

TODO on a future PR:
- Styles for the actual top bar [mentioned here](https://github.com/amaabca/ama_styles/pull/181#pullrequestreview-456104560), not the converted sidebar. (Currently wrong font sizes/colours/arrow colours)

**Pull request merged**